File: RepoTransBench/target_projects/Python/C++/racwn_wp-file-analyser/public_tests/test_public_analyser_branches.cpp
#include <gtest/gtest.h>
#include <string>
#include "wpanalyser/analyser.h"

TEST(PublicAnalyserBranches, DifferentBranchCount) {
    std::string file_content = "<?php\nif(1){echo 'a';}else{echo 'b';}\nif(2){echo 'c';}\n";
    int result = wpanalyser::count_branches(file_content);
    EXPECT_EQ(result, 2);
}

TEST(PublicAnalyserBranches, SwitchCaseBranch) {
    std::string file_content = "<?php\nswitch($var){case 2: break; case 3: break; default: break;}\n";
    int result = wpanalyser::count_branches(file_content);
    EXPECT_EQ(result, 1);
}

TEST(PublicAnalyserBranches, MultipleElseifBranch) {
    std::string file_content = "<?php\nif($a==2){echo 2;}elseif($a==3){echo 3;}elseif($a==4){echo 4;}\n";
    int result = wpanalyser::count_branches(file_content);
    EXPECT_EQ(result, 1);
}

TEST(PublicAnalyserBranches, NestedIfElseBranch) {
    std::string file_content = "<?php\nif($a){if($b){echo 1;}else{echo 2;}}\n";
    int result = wpanalyser::count_branches(file_content);
    EXPECT_EQ(result, 2);
}

TEST(PublicAnalyserBranches, TryCatchCount) {
    std::string file_content = "<?php\ntry{ risky(); } catch(Exception $e) {} catch(Error $e) {}\n";
    int result = wpanalyser::count_branches(file_content);
    EXPECT_EQ(result, 1);
}

File: RepoTransBench/target_projects/Python/Rust/racwn_wp-file-analyser/public_tests/test_public_analyser_branches.rs
use crate::analyser;

#[test]
fn test_different_branch_count() {
    let file_content = "<?php\nif(1){echo 'a';}else{echo 'b';}\nif(2){echo 'c';}\n";
    let result = analyser::count_branches(file_content);
    assert_eq!(result, 2);
}

#[test]
fn test_switch_case_branch() {
    let file_content = "<?php\nswitch($var){case 2: break; case 3: break; default: break;}\n";
    let result = analyser::count_branches(file_content);
    assert_eq!(result, 1);
}

#[test]
fn test_multiple_elseif_branch() {
    let file_content = "<?php\nif($a==2){echo 2;}elseif($a==3){echo 3;}elseif($a==4){echo 4;}\n";
    let result = analyser::count_branches(file_content);
    assert_eq!(result, 1);
}

#[test]
fn test_nested_if_else_branch() {
    let file_content = "<?php\nif($a){if($b){echo 1;}else{echo 2;}}\n";
    let result = analyser::count_branches(file_content);
    assert_eq!(result, 2);
}

#[test]
fn test_try_catch_count() {
    let file_content = "<?php\ntry{ risky(); } catch(Exception $e) {} catch(Error $e) {}\n";
    let result = analyser::count_branches(file_content);
    assert_eq!(result, 1);
}